src/Routing/Route/Route.php

```diff
@@ -146,7 +146,7 @@ public function compiled()
      */
     public function compile()
     {
-        if (!empty($this->compiledRoute)) {
+        if (!empty($this->_compiledRoute)) {
```

Comment:
@chinpei215 You pointed out a valid flaw in the current way of checking. Currently using !empty() is quite dangerous as it would false negative once the variable changes or there is a typo without anyone noticing (assuming there is a missing test case for it).
So it would be sane to start doing this in a way that existing properties will cause an error if you (re)name them wrongly.